### ISSUE TITLE
It would be great to have an option to set the "-parameters" compiler option globally.

### ISSUE THREAD

**Issue by fp024** [redhat-developer/vscode-java#3806]
[provide a description of the issue]

##### Environment
- Operating System: Windows 11
- JDK version: 17 
- Visual Studio Code version:  1.94.0
- Java extension version:  1.35.1

Hello,

I am aware that there is a GUI menu to set the "-parameters".

* https://github.com/redhat-developer/vscode-java/issues/2872

However, when I execute "Clean Java Language Server Workspace", the option gets reset, which feels a bit inconvenient.

So, it would be nice to have an option to set it globally in something like User Settings.json.

Thank you. 👍

**Comment by jdneo** [comment]
Root cause is that by default the .settings folder is stored in the workspace. So it will get removed once the workspace is cleaned.

**Comment by fp024** [comment]
@jdneo 

Thank you for your response.

The reason I submitted the issue was because I frequently ran "Clean Java Language Server Workspace" due to the issue mentioned here: [#3797] 
This reset the settings, so I thought it would be helpful to have a global setting option.

However, while searching, I came across this post: 
* https://github.com/spring-projects/spring-framework/wiki/Upgrading-to-Spring-Framework-6.x#parameter-name-retention.

I tried the following steps:

1. Create the file "${PROJECT_ROOT}/.settings/org.eclipse.jdt.core.prefs". 
  💡 In my case, I've never opened this project to eclipse, so the .settings folder didn't exist.

3. Add the following content to the "org.eclipse.jdt.core.prefs" file:
   ```properties
   org.eclipse.jdt.core.compiler.codegen.methodParameters=generate
   ```

After doing this, the option remains intact even after the completion of running 'Clean Java Language Server Workspace.

I'll close the issue now. Thank you. Have a great day. 👍
